src/impl/icetransport.cpp

```diff
@@ -210,9 +215,17 @@ bool IceTransport::addRemoteCandidate(const Candidate &candidate) {
 	return juice_add_remote_candidate(mAgent.get(), string(candidate).c_str()) >= 0;
 }
 
-void IceTransport::gatherLocalCandidates(string mid) {
+void IceTransport::gatherLocalCandidates(string mid, std::vector<IceServer> additionalIceServers) {
 	mMid = std::move(mid);
 
+	int k = 0;
+	for (auto &server : additionalIceServers) {
```

Comment:
You should randomize `additionalIceServers` before adding, like it is done in the constructor.

Reply:
Done I do a shuffle